docs/design/2023-03-15-ticdc-storage-sink.md

```diff
@@ -0,0 +1,404 @@
+    - `ColumnPrecision`: Column precision. For details, see [Data type](#data-type).
+    - `ColumnScale`: The number of digits following the decimal point (the scale). For details, see [Data type](#data-type).
+    - `ColumnNullable`: The column can be NULL when the value of this option is `true`.
+    - `ColumnIsPk`: The column is part of the primary key when the value of this option is `true`.
+- `TableColumnsTotal`: The size of the `TableColumns` array.
+
+### Data type in schema
+
+This section describes the data types used in the `schema.json` file. The data types are defined as `T(M[, D])`. 
+
+#### Integer types
+
+Integer types in TiDB are defined as `IT[(M)] [UNSIGNED]`, where
+
+- `IT` is the integer type, which can be `TINYINT`, `SMALLINT`, `MEDIUMINT`, `INT`, `BIGINT`, or `BIT`.
+- `M` is the display width of the type.
+
+Integer types are defined as follows in `schema.json`:
+
+```json
+{
+    "ColumnName":"COL1",
+    "ColumnType":"{IT} [UNSIGNED]",
+    "ColumnPrecision":"{M}"
+}
+```
+
+#### Decimal types
+
+Decimal types in TiDB are defined as `DT[(M,D)][UNSIGNED]`, where
+
+- `DT` is the floating-point type, which can be `FLOAT`, `DOUBLE`, `DECIMAL`, or `NUMERIC`.
+- `M` is the precision of the data type, or the total number of digits.
+- `D` is the number of digits following the decimal point.
+
+Decimal types are defined as follows in `schema.json`:
+
+```json
+{
+    "ColumnName":"COL1",
+    "ColumnType":"{DT} [UNSIGNED]",
+    "ColumnPrecision":"{M}",
+    "ColumnScale":"{D}"
+}
+```
+
+#### Date and time types
+
+Date types in TiDB are defined as `DT`, where
+
+- `DT` is the date type, which can be `DATE` or `YEAR`.
+
+The date types are defined as follows in `schema.json`:
+
+```json
+{
+    "ColumnName":"COL1",
+    "ColumnType":"{DT}"
+}
+```
+
+The time types in TiDB are defined as `TT[(M)]`, where
+
+- `TT` is the time type, which can be `TIME`, `DATETIME`, or `TIMESTAMP`.
+- `M` is the precision of seconds in the range from 0 to 6.
+
+The time types are defined as follows in `schema.json`:
+
+```json
+{
+    "ColumnName":"COL1",
+    "ColumnType":"{TT}",
+    "ColumnScale":"{M}"
+}
+```
+
+#### String types
+
+The string types in TiDB are defined as `ST[(M)]`, where
+
+- `ST` is the string type, which can be `CHAR`, `VARCHAR`, `TEXT`, `BINARY`, `BLOB`, or `JSON`.
+- `M` is the maximum length of the string.
+
+The string types are defined as follows in `schema.json`:
+
+```json
+{
+    "ColumnName":"COL1",
+    "ColumnType":"{ST}",
+    "ColumnLength":"{M}"
+}
+```
+
+#### Enum and Set types
+
+The Enum and Set types are defined as follows in `schema.json`:
+
+```json
+{
+    "ColumnName":"COL1",
+    "ColumnType":"{ENUM/SET}",
+}
+```
+
+### Protocols
+
+#### CSV
+
+##### Transactional constraints
+- In a single CSV file, the commit-ts of a row is equal to or smaller than that of the subsequent row.
+- The same transactions of a single table are stored in the same CSV file.
```

Comment:
Is this suitable for large transaction?

Reply:
Good advice, the above conditions can be guaranteed only when transaction-atomicity is set to table level.